### PR TITLE
Add CI slack notifications.

### PR DESCRIPTION
**Pull request by alliepiper**
The notifications are just for nightlies/weeklies/etc.

There will be two slack channels -- one that only reports failures, and one that reports everything. The first is meant to be monitored by the team so we know when things break, the other is more of a log for looking at historical results, runtimes, detecting startup issues, etc.

### CODE REVIEW

File: .github/actions/workflow-build/action.yml [hunk 1]
```diff
@@ -35,6 +44,20 @@ runs:
...
+        SLACK_BOT_TOKEN: ${{ inputs.slack_token }}
+        WORKFLOW_TYPE: ${{ github.workflow }} # nightly, weekly, pr, etc.
+        SUMMARY_URL: https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+      with:
+        channel-id: ${{ inputs.slack_log }}
+        slack-message: |
+          Workflow '${{ env.WORKFLOW_TYPE }}' starting...
+
+          Details: ${{ env.SUMMARY_URL }}
+
```
Comment (bernhardmgruber) [inline .github/actions/workflow-build/action.yml:60]: Question: This sends a notification every time a workflow is starting. Isn't this a lot of noise? And who is interested in these kinds of notifications? I also know a workflow is starting when I create a PR or when I add a comment `/ok to test`. What problem is this trying to solve?

**Comment by jrhemstad** [comment]
@alliepiper could you update the PR description to summarize when notifications will be pushed to various slack channels? Like @bernhardmgruber I want to make sure we're starting conservatively such that we have a meaningful signal-to-noise ratio.

File: .github/actions/workflow-build/action.yml (hunk 1 above)
Comment (alliepiper) [reply .github/actions/workflow-build/action.yml:60]: The notifications are just for nightlies/weeklies/etc, not per PR. Right now the nightly failures aren't reported anywhere.

There will be two slack channels -- one that only reports failures, and one that reports everything. The first is meant to be monitored by the team so we know when things break, the other is more of a log for looking at historical results, runtimes, startup issues, etc.

**Comment by alliepiper** [comment]
@jrhemstad done

File: .github/actions/workflow-build/action.yml [hunk 2]
```diff
@@ -25,6 +25,15 @@ inputs:
...
     required: false
+  slack_token:
+    description: "The Slack token to use for notifications. No notifications will be sent if not provided."
+    required: false
+  slack_log:
+    description: "Slack channel ID for verbose notifications."
+    required: false
+  slack_alert:
+    description: "Slack channel ID for alert notifications."
+    required: false
```
Comment (alliepiper) [inline .github/actions/workflow-build/action.yml:36]: @jrhemstad The slack notifications are only sent when these are defined.

File: .github/workflows/ci-workflow-nightly.yml [hunk 3]
```diff
@@ -128,6 +131,11 @@ jobs:
       - name: Check workflow success
         id: check-workflow
         uses: ./.github/actions/workflow-results
+        with:
+          slack_token: ${{ secrets.SLACK_NOTIFIER_BOT_TOKEN }}
+          slack_log: ${{ secrets.SLACK_CHANNEL_CI_LOG }}
+          slack_alert: ${{ secrets.SLACK_CHANNEL_CI_ALERT }}
```
Comment (alliepiper) [inline .github/workflows/ci-workflow-nightly.yml:137]: @jrhemstad We only define these here for the nightly workflow.

File: .github/actions/workflow-build/action.yml (hunk 1 above)
Comment (bernhardmgruber) [reply .github/actions/workflow-build/action.yml:60]: I didn't see it was intended for the nightlies. I also perceive having a history of testing runtimes very useful. That sounds a lot more convincing! Thx!

**Review by jrhemstad — APPROVED** [review]

**Pull request merged**